server/ip-tools.ts

```diff
@@ -129,20 +133,20 @@ export const IPTools = new class {
 			num = (num - part) / 256;
 			ipParts.unshift(part.toString());
 		}
+		if (ipParts.length !== 4) return null;
```

Comment:
![image](https://user-images.githubusercontent.com/551184/119421953-be6ece00-bcb4-11eb-8d3b-0197a3569890.png)

```suggestion
		while (ipParts.length < 4) ipParts.unshift(0);
		if (ipParts.length !== 4) return null;
```